magefiles/rulesengine/repos/infra_deployments.go

```diff
@@ -1,17 +1,158 @@
-			rctx.LabelFilter = "e2e-demo,konflux-demo,spi-suite,remote-secret,integration-service,ec,build-templates,multi-platform"
-			return ExecuteTestAction(rctx)
-		})}},
+import (
+	"fmt"
+	"strings"
+
+	"github.com/konflux-ci/e2e-tests/magefiles/rulesengine"
+)
+
+// Default Rule of repo infra-deployments running konflux-demo suite.
+var InfraDeploymentsDefaultTestRules = rulesengine.Rule{Name: "Infra Deployments Default Test Execution",
+	Description: "Run the Konflux-demo suite tests when an Infra-deployments PR includes changes to files outside of the specified components.",
+	Condition: rulesengine.None{
+		&InfraDeploymentsIntegrationTestFileChangeRule,
+		&InfraDeploymentsImageControllerTestFileChangeRule,
+		&InfraDeploymentsMultiPlatformTestFileChangeRule,
+		&InfraDeploymentsBuildTemplatesTestFileChangeRule,
+		&InfraDeploymentsBuildServiceTestFileChangeRule,
+		&InfraDeploymentsReleaseServiceTestFileChangeRule,
+		&InfraDeploymentsEnterpriseControllerTestFileChangeRule,
+		&InfraDeploymentsJVMTestFileChangeRule,
+		rulesengine.ConditionFunc(CheckNoFilesChanged)},
+
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(ExecuteInfraDeploymentsDefaultTestAction)}}
+
+// ExecuteInfraDeploymentsDefaultTestAction excutes all the e2e-tests and component suites
+func ExecuteInfraDeploymentsDefaultTestAction(rctx *rulesengine.RuleCtx) error {
+	rctx.LabelFilter = "konflux-demo"
+	return ExecuteTestAction(rctx)
 }
+
+// InfraDeploymentsTestFilesOnlyRule defines rules of test suites running of each changed component
+var InfraDeploymentsTestFilesOnlyRule = rulesengine.Rule{Name: "Infra-deployments PR Components File Diff Execution",
+	Description: "Runs specific tests of changed component by infra-deployments PR.",
+	Condition: rulesengine.Any{
+		&InfraDeploymentsIntegrationTestFileChangeRule,
+		&InfraDeploymentsImageControllerTestFileChangeRule,
+		&InfraDeploymentsMultiPlatformTestFileChangeRule,
+		&InfraDeploymentsBuildTemplatesTestFileChangeRule,
+		&InfraDeploymentsBuildServiceTestFileChangeRule,
+		&InfraDeploymentsReleaseServiceTestFileChangeRule,
+		&InfraDeploymentsEnterpriseControllerTestFileChangeRule,
+		&InfraDeploymentsJVMTestFileChangeRule},
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		// Adding konflux-demo to the label filtetr when componenets changed
+		rctx.LabelFilter = fmt.Sprintf("%s,%s", rctx.LabelFilter, "konflux-demo")
+		return nil
+
+	}),
+		rulesengine.ActionFunc(ExecuteTestAction)}}
+
+var InfraDeploymentsIntegrationTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Integration component File Change Rule",
+	Description: "Map Integration tests files when Integration component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/integration/*")) != 0
```

Comment:
I think this expression covers just **files** under `components/integration/` dir. We should be covering [all directories and files under it as well](https://github.com/redhat-appstudio/infra-deployments/tree/main/components/integration)

Reply:
I made a change into a yaml file in **components/integration/development/kustomization.yaml** , commit 
then ran the demo ,  looks like the change was identified : 

```
exec: git "-C" "/home/kasemalem/integration-project/e2e-tests/tmp/infra-deployments" "diff" "--name-status" "upstream/main"
I0816 11:31:19.981602 1290396 utils.go:473] The following files, components/integration/OWNERS, components/integration/development/kustomization.yaml, were changed!
I0816 11:31:19.981659 1290396 types.go:72] Loading the catalog for, infra-deployments, from category, tests
I0816 11:31:19.981678 1290396 types.go:144] The following rules have matched Infra-deployments PR Components File Diff Execution.
I0816 11:31:19.981684 1290396 types.go:157] DryRun has been enabled will apply them in dry run mode
exec: ginkgo "--seed=1723797078" "--dry-run" "--timeout=1h30m0s" "--grace-period=30s" "--output-interceptor-mode=none" "--label-filter=integration-service,konflux" "--junit-report=e2e-report.xml" "--p" "--output-dir=." "./cmd" "--"
```
 this means that current rule will be working also of changes made recursively, @psturc WDYT?

Reply:
I think it was identified only because it contained the change to `components/integration/OWNERS`

Reply:
I just tested it with an update to build service and it doesn't work
```
➜  e2e-tests git:(KONFLUX-3187) ./mage -v local:runruledemo
Running target: Local:RunRuleDemo
exec: git "-C" "/Users/psturc/work/konflux-ci/e2e-tests/tmp/infra-deployments" "diff" "--name-status" "upstream/main"
I0816 12:19:24.863695   60516 utils.go:473] The following files, components/build-service/base/kustomization.yaml, were changed!
I0816 12:19:24.863824   60516 types.go:72] Loading the catalog for, infra-deployments, from category, tests
I0816 12:19:24.863846   60516 types.go:144] The following rules have matched Infra Deployments Default Test Execution.
I0816 12:19:24.863858   60516 types.go:169] Will apply rules
exec: ginkgo "--seed=1723803564" "--timeout=1h30m0s" "--grace-period=30s" "--output-interceptor-mode=none" "--label-filter=konflux" "./cmd" "--"
```